repository/src/main/java/org/alfresco/repo/content/replication/AggregatingContentStore.java

```diff
@@ -448,4 +442,97 @@ else if (!contentUrlSupported)
+    {
+        checkPrimaryStore();
+        // get a read lock so that we are sure that no replication is underway
+        readLock.lock();
+        boolean archiveRequestSucceeded = false;
+        boolean primaryContentUrlUnsupported = false;
+        boolean secondaryContentUrlUnsupported = false;
+        try {
+            // Check the primary store
+            try {
+                archiveRequestSucceeded = hasArchiveRequestSucceeded(contentUrl, restoreParams, restore, primaryStore);
+            } catch (UnsupportedOperationException e) {
+                final String message = String.format("Primary store does not handle this operation for content URL: %s", contentUrl);
+                logTrace(message);
+            } catch (UnsupportedContentUrlException e) {
+                final String message = String.format(PRIMARY_STORE_COULD_NOT_HANDLE_CONTENT_URL, contentUrl);
+                logTrace(message);
+                primaryContentUrlUnsupported = true;
+            }
+
+            if (archiveRequestSucceeded) {
+                return true;
+            } else { // the content is not in the primary store so we have to go looking for it
+                for (ContentStore store : secondaryStores) {
+                    try {
+                        archiveRequestSucceeded = hasArchiveRequestSucceeded(contentUrl, restoreParams, restore, store);
+                    } catch (UnsupportedOperationException e) {
+                        final String message =
+                                String.format("Secondary store %s does not handle this operation for content URL: %s", store,
+                                        contentUrl);
+                        logTrace(message);
+                    } catch (UnsupportedContentUrlException e) {
+                        secondaryContentUrlUnsupported = true;
+                        final String message = String.format(SECONDARY_STORE_COULD_NOT_HANDLE_CONTENT_URL, store, contentUrl);
+                        logTrace(message);
+                    }
+                }
+            }
+            if (archiveRequestSucceeded) {
+                return true;
+            } else if (primaryContentUrlUnsupported || secondaryContentUrlUnsupported) {
+                return super.requestSendContentToArchive(contentUrl);
+            }
+
+            return super.requestSendContentToArchive(contentUrl);
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    private boolean hasArchiveRequestSucceeded(String contentUrl, Map<String, Serializable> restoreParams, boolean restore,
+                                               ContentStore primaryStore)
+    {
+        return restore ?
+                primaryStore.requestRestoreContentFromArchive(contentUrl, restoreParams) :
+                primaryStore.requestSendContentToArchive(contentUrl);
+    }
+
+    private void logTrace(final String message)
```

Comment:
Minor: Is this extra method needed ? Could you use log.trace("my message %s", mystr) instead (in most case cases) ?

Reply:
Thanks, switched to `logger.trace(message)`